File: C++/C++Basic/06.Conditional-Statements-Advanced-Exercise/09.skiTrip.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    int days;
    string typeOfRoom, feedback;

    cin >> days;
    cin.ignore();

    getline(cin, typeOfRoom);
    getline(cin, feedback);

    double price = 0.0;

    if (typeOfRoom == "room for one person")
    {
        price = 18;
    }
    else if (typeOfRoom == "apartment")
    {
        price = 25;
    }

    else if (typeOfRoom == "president apartment")
    {
        price = 35;
    }

    int numberOfNight = days - 1;
    double totalPrice = numberOfNight * price;

    if (typeOfRoom == "apartment")
    {
        if (days < 10)
        {
            totalPrice *= 0.70;
        }
        else if (10 <= days && days <= 15)
        {
            totalPrice *= 0.65;
        }
        else if (days > 15)
        {
            totalPrice *= 0.50;
        }
    }
    else if (typeOfRoom == "president apartment")
    {
          if (days < 10)
        {
            totalPrice *= 0.90;
        }
        else if (10 <= days && days <= 15)
        {
            totalPrice *= 0.85;
        }
        else if (days > 15)
        {
            totalPrice *= 0.80;
        }
    }

    if (feedback == "positive"){
        totalPrice *= 1.25;
    }
    else
    {
        totalPrice *= 0.90;
    }

    cout.setf(ios::fixed);
    cout.precision(2);

    cout << totalPrice << endl;

    return 0;
}

File: Python/Basic/06.Conditional-Statements-Advanced-Exercise/09.ski_trip.py
days = int(input())
type_of_room = input()
feedback = input()

days -= 1
price = 0

if type_of_room == "room for one person":
    price = 18.00 * days

elif type_of_room == "apartment":
    price = 25.00 * days

    if days < 10:
        price *= 0.70
    elif 10 <= days <= 15:
        price *= 0.65
    elif days > 15:
        price *= 0.50

elif type_of_room == "president apartment":
    price = 35.00 * days

    if days < 10:
        price *= 0.90
    elif 10 <= days <= 15:
        price *= 0.85
    elif days > 15:
        price *= 0.80

if feedback == "positive":
    price *= 1.25
else:
    price *= 0.90

print(f"{price:.2f}")